src/ar_http_iface_server.erl

```diff
@@ -880,337 +33,57 @@ reregister(Name, Node) ->
-block_field_to_string(<<"hash">>, Res) -> Res;
-block_field_to_string(<<"indep_hash">>, Res) -> Res;
-block_field_to_string(<<"txs">>, Res) -> ar_serialize:jsonify(Res);
-block_field_to_string(<<"hash_list">>, Res) -> ar_serialize:jsonify(Res);
-block_field_to_string(<<"wallet_list">>, Res) -> ar_serialize:jsonify(Res);
-block_field_to_string(<<"reward_addr">>, Res) -> Res.
-
-%% @doc checks if hash is valid & if so returns filename.
-hash_to_filename(Type, Hash) ->
-	case safe_decode(Hash) of
-		{error, invalid} ->
-			{error, invalid};
-		{ok, ID} ->
-			{Mod, Fun} = type_to_mf({Type, lookup_filename}),
-			F = apply(Mod, Fun, [ID]),
-			case F of
-				unavailable ->
-					{error, ID, unavailable};
-				Filename ->
-					{ok, Filename}
-			end
-	end.
-
-%% @doc Return true if ID is a pending tx.
-is_a_pending_tx(ID) ->
-	lists:member(ID, ar_node:get_pending_txs(whereis(http_entrypoint_node))).
-
-%% @doc Given a request, returns a blockshadow.
-request_to_struct_with_blockshadow(Req) ->
-	try
-		BlockJSON = elli_request:body(Req),
-		{Struct} = ar_serialize:dejsonify(BlockJSON),
-		JSONB = val_for_key(<<"new_block">>, Struct),
-		BShadow = ar_serialize:json_struct_to_block(JSONB),
-		{ok, {Struct, BShadow}}
-	catch
-		Exception:Reason ->
-			{error, {Exception, Reason}}
-	end.
-
-%% @doc Generate and return an informative JSON object regarding
-%% the state of the node.
-return_info() ->
-	{Time, Current} =
-		timer:tc(fun() -> ar_node:get_current_block_hash(whereis(http_entrypoint_node)) end),
-	{Time2, Height} =
-		timer:tc(fun() -> ar_node:get_height(whereis(http_entrypoint_node)) end),
-	{200, [],
-		ar_serialize:jsonify(
-			{
-				[
-					{network, list_to_binary(?NETWORK_NAME)},
-					{version, ?CLIENT_VERSION},
-					{release, ?RELEASE_NUMBER},
-					{height,
-						case Height of
-							not_joined -> -1;
-							H -> H
-						end
-					},
-					{current,
-						case is_atom(Current) of
-							true -> atom_to_binary(Current, utf8);
-							false -> ar_util:encode(Current)
-						end
-					},
-					{blocks, ar_storage:blocks_on_disk()},
-					{peers, length(ar_bridge:get_remote_peers(whereis(http_bridge_node)))},
-					{queue_length,
-						element(
-							2,
-							erlang:process_info(whereis(http_entrypoint_node), message_queue_len)
+%% @doc Start the server
+do_start(Port, TransportOpts) ->
+	spawn(
+		fun() ->
+			ar_blacklist:start(),
+			Routes = [
+				{'_', [
+					{"/metrics/[:registry]", prometheus_cowboy2_handler, []},
+					{"/[...]", ar_http_iface_cowboy_handler, []}
+				]}
+			],
+			Dispatch = cowboy_router:compile(Routes),
+			ProtocolOpts =
+				#{
+					middlewares => [
+						ar_blacklist,
+						cowboy_router,
+						cowboy_handler
+					],
+					env => #{dispatch => Dispatch},
+					metrics_callback => fun prometheus_cowboy2_instrumenter:observe/1,
+					stream_handlers => [cowboy_metrics_h, cowboy_stream_h]
+				},
+
+			case TransportOpts of
+				clear ->
```

Comment:
How do you feel about calling this `no_tls`?

Reply:
I took the wording from cowboy's `start_clear`/`start_tls` but I can't find other languages/libraries that refer to non-secure HTTP as clear... I'm up for any wording